### ISSUE TITLE
[no-duplicate-imports] False positive in d.ts file with multiple module declarations

### ISSUE THREAD

**Issue by danielrentz** [typescript-eslint/typescript-eslint#3071]
The rule triggers on multiple imports inside module declarations in the same d.ts file.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

.eslintrc.yaml
```YAML
overrides:
- files: ["**/*.ts"]
  rules:
    no-duplicate-imports: off
    "@typescript-eslint/no-duplicate-imports": error
```

core-modules.d.ts
```TS
declare module "core/module1" {
    export interface I { }
}

declare module "core/module2" {
    import { I } from "core/module1";
    export const i1: I;
}

declare module "core/module3" {
    import { I } from "core/module1";
    export const i2: I;
}
```

**Expected Result**
The d.ts file should pass without error.

**Actual Result**
Second import of "core/module1" (inside "core/module3") triggers the error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `14.5.1` |
| `@typescript-eslint/parser`        | `14.5.1` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.20.0` |
| `node`                             | `12.16.2` |

**Comment by arkajyotiMukherjee** [comment]
same issue :)

**Comment by Josh-Cena** [comment]
Now that `no-duplicate-imports` is deprecated, I think this can be closed?
